src/evaluators/JSXElement.js

```diff
@@ -272,20 +266,22 @@ function evaluateJSXAttributes(
     }
   }
 
-  if (containsAbstractSpreadAttribute) {
+  if (abstractSpreadCount > 0) {
     // if we haven't assigned any attributes and we are dealing with a single
     // spread attribute, we can just make the spread object the props
     if (
       attributesAssigned === 0 &&
       ((spreadValue instanceof ObjectValue && spreadValue.isPartialObject()) || spreadValue instanceof AbstractValue)
     ) {
-      // the spread is partial, so we can re-use that value
-      config = spreadValue;
-      if (config instanceof ObjectValue || config instanceof AbstractObjectValue) {
-        // as we're applying a spread, the config needs to be simple/partial
+      if (spreadValue instanceof AbstractValue && !(spreadValue instanceof AbstractObjectValue)) {
+        config = To.ToObject(realm, spreadValue);
+      } else {
+        // the spread is partial, so we can re-use that value
+        config = spreadValue;
         config.makePartial();
         config.makeSimple();
       }
+      resetRefAndKeyFromProps(realm, config);
```

Comment:
If config is an abstract coming from external code why is it safe to assume it has no ref and key? I don't understand this.

I understand why `props` wouldn't have ref/key on it. But not why `config` wouldn't have it. As far as I can tell `spreadValue` could literally be anything.